### ISSUE TITLE
--hide-timeout used with --output json --static produces unparseable JSON

### ISSUE THREAD

**Issue by cralston0** [saltstack/salt#40311]
### Description of Issue/Question

When using ``--output json --static``, the option``--hide-timeout`` with any minions offline produces a message that makes the output illegal JSON.

The info ``Minion NAME did not respond. No job will be sent.`` should probably either be sent to stderr instead of stdout, or it shouldn't be output at all when ``--hide-timeout`` is used.

Further experimenting reveals this only happens when ``--static`` is also used.  Without ``--static``, the offline minions are properly skipped with no warning/error.

### Setup

No special setup needed -- A master with one minion that is offline can duplicate.

### Steps to Reproduce Issue

```
$ salt --hide-timeout --static --output json '*' cmd.run 'uptime'
Minion foobar did not respond. No job will be sent.
{
    "widget": {
        "jid": "20170324182543990496", 
        "retcode": 0, 
        "ret": "..."
    }, 
...
$ salt --hide-timeout --static --output json '*' cmd.run 'uptime' 2>/dev/null
Minion foobar did not respond. No job will be sent.
{
    "widget": {
        "jid": "20170324182543990496", 
        "retcode": 0, 
        "ret": "..."
    }, 
...
```

### Versions Report
```
Salt Version:
           Salt: 2016.11.1
 
Dependency Versions:
           cffi: 1.10.0
       cherrypy: Not Installed
       dateutil: 2.6.0
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 0.9.1
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: 1.2.3
      pycparser: 2.17
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.6 (default, Oct 26 2016, 20:30:19)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 14.0.1
           RAET: Not Installed
          smmap: 0.8.2
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
 
System Versions:
           dist: Ubuntu 14.04 trusty
        machine: x86_64
        release: 3.13.0-105-generic
         system: Linux
        version: Ubuntu 14.04 trusty
```

**Comment by gtmanfred** [comment]
That message should be changed to be sent to stderr.  Thanks for reporting.  If you wanted to submit a PR to fix this, we would greatly appreciate it, otherwise we will get it fixed.

Thanks,
Daniel

**Comment by DenisBY** [comment]
I have similar issue:
```
# salt -l quiet --out=json --static -G "roles:salt-master" cmd.run "hostname" 2>/dev/null
Minion i-077a6cd7d61696325 did not respond. No job will be sent.
Minion i-0750f7677157015c0 did not respond. No job will be sent.
Minion i-07561c1acac80c6c0 did not respond. No job will be sent.
Minion i-0b32a8ab83e7f0f47 did not respond. No job will be sent.
Minion i-44b4f885 did not respond. No job will be sent.
Minion i-0dbf64eddb369dd6d did not respond. No job will be sent.
Minion i-040c5d3d1f5472fca did not respond. No job will be sent.
Minion i-08b120096781b8209 did not respond. No job will be sent.
Minion i-0a444eac606f6f42c did not respond. No job will be sent.
Minion i-0592d9d9a42d49c82 did not respond. No job will be sent.
Minion i-042c0213d9ec4d8ea did not respond. No job will be sent.
{
    "salt-master": "salt-master"
}
```

How to suppress `Minion did not respond. No job will be sent.` output?

**Comment by gtmanfred** [comment]
pass it threw awk, until this bug is fixed?

`salt -l quiet --out=json --static -G "roles:salt-master" cmd.run "hostname" 2>/dev/null | awk '!/^Minion/'`

**Comment by DenisBY** [comment]
I'm doing it via `grep -v ^Minion`. Didn't have this issue with 2016.3.x

**Comment by gtmanfred** [comment]
This will be fixed with the above PR.

**Comment by DenisBY** [comment]
Thanks. Patched it manually.

**Comment by DenisBY** [comment]
Will this fix be included in 2017.7.2 as well?

**Comment by gtmanfred** [comment]
I do not believe so, but it will be in 2017.7.3
